tutorials/3d/csg_tools.rst

```diff
@@ -0,0 +1,191 @@
+
+Create 4 ``CSGBox`` children nodes in ``union`` mode for the legs and adjust the dimensions.
+
+Adjust the position to form a desk.
+
+.. image:: img/csg_desk.png
+
+.. note:: CSG nodes inside a ``CSGCombiner`` will only process their operation within the combiner, therefore, ``CSGCombiners`` are used to organize the CSG nodes. 
+
+Create a ``CSGCombiner`` and name it ``bed``.
+
+Our bed consists of 3 parts: the bed, the mattress, and a pillow.
+
+Create a ``CSGBox`` and adjust its dimension for the bed. Create another ``CSGBox`` and adjust its dimension for the mattress.
+
+.. image:: img/csg_bed_mat.png
+
+We will create another ``CSGCombiner`` and name it ``pillow`` as the child of  ``bed``. The scene tree should look like this:
+
+.. image:: img/csg_bed_tree.png
+
+We will combine 3 ``CSGshpere`` in ``union`` mode to form a pillow. Scale the y-axis of the spheres and enable ``smooth faces``.
+
+.. image:: img/csg_pillow_smooth.png
+
+Select the `pillow` node and switch the mode to ``subtraction``, the combined spheres will cut a hole on the mattress.
+
+.. image:: img/csg_pillow_hole.png
+
+Try to re-parent the ``pillow`` node to the root ``Spatial`` node, the hole will disappear.
+
+.. note:: this is to illustrate the effect of CSG process order. Since the root node is not a CSG node, the ``CSGCombiner`` nodes are the end of the operations, this shows the use of ``CSGCombiner`` to organize the CSG scene. 
+
+Undo the re-parent after observing the effect.
+
+The bed you built should look like this:
+
+.. image:: img/csg_bed.png
+
+
+Create a ``CSGCombiner`` and name it ``lamp``.
+
+A lamp consists of 3 parts: the stand, the pole, and the lampshade.
+
+Create a ``CSGCylinder`` enable option ``cone`` and make it the stand. Create another ``CSGCylinder`` and adjust the dimensions to use it as a pole.
+
+.. image:: img/csg_lamp_pole_stand.png
+
+We will use ``CSGPolygon`` for the lampshade. Use ``Spin`` mode for the ``CSGPolygon`` and draw a trapezium in front view, this shape will extrude around the origin and form the lampshade.
+
+.. image:: img/csg_lamp_spin.png
+
+.. image:: img/csg_lamp_polygon.png
+
+.. image:: img/csg_lamp_extrude.png
+
+Adjust the position of the 3 parts to make it a ``lamp``.
+
+.. image:: img/csg_lamp.png
+
+Create a ``CSGCombiner`` and name it ``bookshelf``.
+
+We will use 3 ``CSGBox`` for the bookshelf. Create a ``CSGBox`` and adjust its dimension, this will be the size of the bookshelf. 
+
+.. image:: img/csg_shelf_big.png
+
+Duplicate the ``CSGBox`` and shorten the dimension of each axis and change the mode to ``Subtraction``. 
+
+.. image:: img/csg_shelf_subtract.png
+
+.. image:: img/csg_shelf_subtract_menu.png
+
+You almost build a shelf. Create one more ``CSGBox`` for dividing the shelf into two levels.
+
+.. image:: img/csg_shelf.png
+
+Position your furniture in your room as you like and your scene should look this:
+
+.. image:: img/csg_room_result.png
+
```

Comment:
This ends very abruptly IMO. Add a ending sentence congratulating on finishing the tutorial?